AxonDeepSeg/segment.py

```diff
@@ -169,9 +233,36 @@ def segment_folders(path_testing_images_folder, path_model,
 
         acquisition_name = file_.name
        
-        axon_segmentation(path_acquisitions_folders=path_testing_images_folder, acquisitions_filenames=[str(path_testing_images_folder  / acquisition_name)],
-                  path_model_folder=path_model, overlap_value=overlap_value,
-                  acquired_resolution=acquired_resolution)
+
+        # Check that the resampled image will be of sufficient size, and if not throw an error.
+        h, w = image_size
+
+        w_resampled = w*(acquired_resolution*zoom_factor)/model_resolution
+        h_resampled = h*(acquired_resolution*zoom_factor)/model_resolution
+
+        if w_resampled < patch_size or h_resampled < patch_size:
+            if w<=h:
+                minimum_zoom_factor = patch_size*model_resolution/(w*acquired_resolution)
+            else:
+                minimum_zoom_factor = patch_size*model_resolution/(h*acquired_resolution)
+
+            # Round to 1 decimal, always up.
+            minimum_zoom_factor = ceil(minimum_zoom_factor*10)/10
+
+            print("ERROR: Due to your given image size, resolution, and zoom factor, at least one image, ", 
+                   str(path_testing_images_folder / file_),
+                   ", is smaller than",
+                   "the patch size after it is resampled during segmentation. To resolve this, please set a zoom factor greater than ",
+                   str(minimum_zoom_factor), ".",
+                   "To do this on the command line, call the segmentation with the -z flag, i.e. ",
+                   "-z ", str(minimum_zoom_factor),
+            )
+            sys.exit(4)
```

Comment:
This check is done in the for loop, so the error will be raised only when 1 of the file fails, after segmenting previous X images that were OK and leaving the next Y images not segmented.

Suggestions:
1. Check the whole folder for "bad" resolution before starting the segmentation.
2. Or skip to the next file and provide a list of "failed" segmentation at the end of the folder.

Reply:
I think we need to do option 2, as we would have to load each images once just to do 1), and then reload them again when we actually do the segmentation.

Should we start to have a log for these types of situations?

Reply:
Yes, a log is a good idea!
Saving the terminal output in a file (with date and time in the name) along with the segmentations would be helpful.

This may be out of scope for this PR. The current behavior is not ideal, but I would be comfortable merging like this and dealing with the loop and log separately if you prefer.